cells/lib/ops/reviseOCI.nix

```diff
@@ -0,0 +1,63 @@
+{
+  inputs,
+  cell,
+}: let
+  inherit (inputs) nixpkgs;
+  l = nixpkgs.lib // builtins;
+in
+  /*
+  Utility function to allow for building containers in a mono-repo style environment where
+  the source code is contained in the same repository as the std code, specifically so that
+  one may detect meaningful changes to the image via its tag in the special case where the
+  package's output includes the revision of the source code (e.g. for displaying the version
+  to the user).
+
+  Without special processing, this kind of package would cause the OCI image tag to change
+  on each new revision whether the actual contents of the image changed or not. Combined
+  with `std.incl`, one may have a very strong indicator for when the contents of the image
+  actually includes meaningful changes which avoids flooding the remote registry with superflous
+  copies.
+
+  This function can also be called where the package does not need the revsion at build time
+  but you simply want to tag the image by its hash for later processing by the proviso, and
+  you also want to include additional tags on the image, such as the revision.
+
+  Args:
+  args: arguments to the mkOCI function to be called.
+  mkOCI: defaults to `mkStandardOCI`
+  operable: The operable to include in the container
+  ...: The same arguments expected by the given standard OCI builder
+
+  Returns:
+  An image tagged with the output hash of an identical image, except where the target package
+  and operable are both built with the revision input set to "not-a-commit" instead of the true
+  revision, so that the hash does not change unless something inside the image does.
+  */
+  args @ {
+    operable,
+    mkOCI ? cell.ops.mkStandardOCI,
+    ...
+  }: let
+    revision = cell.ops.revise mkOCI args.operable (operable: builtins.removeAttrs (args // {inherit operable;}) ["mkOCI"]);
+  in
+    if args.operable ? sansrev
+    then
+      mkOCI (args
+        // {
+          meta =
+            args.meta
+            or {}
+            // {
+              tags = [revision.sansrev.outHash] ++ (args.meta.tags or []);
+            };
+        })
+    else
+      mkOCI (args
+        // {
+          meta =
+            args.meta
+            or {}
+            // {
+              tags = [(cell.ops.hashOfPath revision.outPath)] ++ (args.meta.tags or []);
```

Comment:
Hm, now I wonder even more how the magic of `sansrev` signalling and dry-running works.

Maybe `revise.nix` would indeed be a good place to explain.

Reply:
The whole point of sansrev is to have a version of the package that doesn't change when the revision does. The only way to guarantee this is to take the rev as an argument to callPackage (hence why `rev` is a required argument to `revisePackage`), and pass in the default `not-a-commit` unconditionally to it.

I tried more clever approaches like modifying `src.origSrc.rev` but the hash of the outPath hash of the src is different so that didn't work.

Reply:
Thanks, the overall strategy is clear to me _now_, but I'm more advocating for code comments on behalf of a future self or present/future other.

Can you curry and spice up the code with a couple of code comments that guide the _code_ reader through the logic of `sansrev`?